core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -39,10 +38,11 @@ object RequestChannel extends Logging {
   val AllDone = new Request(processor = 1, connectionId = "2", new Session(KafkaPrincipal.ANONYMOUS, InetAddress.getLocalHost()), buffer = getShutdownReceive(), startTimeMs = 0, securityProtocol = SecurityProtocol.PLAINTEXT)
 
   def getShutdownReceive() = {
-    val emptyProducerRequest = new ProducerRequest(0, 0, "", 0, 0, collection.mutable.Map[TopicAndPartition, ByteBufferMessageSet]())
-    val byteBuffer = ByteBuffer.allocate(emptyProducerRequest.sizeInBytes + 2)
-    byteBuffer.putShort(ApiKeys.PRODUCE.id)
-    emptyProducerRequest.writeTo(byteBuffer)
+    val emptyRequestHeader = new RequestHeader(ApiKeys.PRODUCE.id, "", 0)
+    val emptyProduceRequest = new ProduceRequest(0, 0, new HashMap[TopicPartition, ByteBuffer]())
+    val byteBuffer = ByteBuffer.allocate(emptyRequestHeader.sizeOf() + emptyProduceRequest.sizeOf())
```

Comment:
Not critical, but could we reuse `RequestSend.serialize()` here and avoid duplicating low-level serialization code?